airgun/widgets.py

```diff
@@ -916,31 +916,44 @@ class CustomParameter(Table):
 
     add_new_value = Text("..//a[contains(text(),'+ Add Parameter')]")
 
-    def __init__(self, parent, locator=None, id=None, logger=None):
+    # def __init__(self, parent, locator=None, id=None, logger=None, column_widgets=None):
+    def __init__(self, parent, **kwargs):
         """Supports initialization via ``locator=`` or ``id=``"""
-        if locator and id or not locator and not id:
+        if (
+            kwargs.get('locator')
+            and kwargs.get('id')
+            or not kwargs.get('locator')
+            and not kwargs.get('id')
+        ):
             raise ValueError('Please specify either locator or id')
-        locator = locator or f".//table[@id='{id}']"
+        locator = kwargs.get('locator') or f".//table[@id='{kwargs.get('id')}']"
 
-        column_widgets = {
+        column_widgets = kwargs.get('column_widgets') or {
             'Name': TextInput(locator=".//input[@placeholder='Name']"),
             'Value': TextInput(locator=".//textarea[@placeholder='Value']"),
             'Actions': Text(
                 locator=".//a[@data-original-title='Remove Parameter' "
                 "or @title='Remove Parameter']"
             ),
         }
-        super().__init__(parent, locator=locator, logger=logger, column_widgets=column_widgets)
+        self.name = list(column_widgets.keys())[0]
```

Comment:
This could throw `IndexError` during instantiation if column_widgets weren't passed.

Have you checked all uses of this widget class against this requirement?

I think you'll want to make this safer by having some checking, or using try/except to set these instance attributes to `None` if column_widgets doesn't have the expected length/field format.